docs/org.eclipse.n4js.spec/chapters/05_classifiers/classifiers-part-02-members.adoc

```diff
@@ -639,41 +639,138 @@ class B extends A {
-For more details about this typing can be found in <<_structural-read-only-write-only-and-initializer-field-typing>>.
-This can be used as shown in the following listing:
+The tedious process of copying the members of the parameter to the fields of the class can be automated via the `@Spec`
+annotation if the argument has `pass:[~i~this]` structural initializer field typing. More details about this typing can be
+found in <<structural-readWriteInit-field-typing>>. This can be used as shown in the following listing:
 
 [source,n4js]
 ----
 class A {
+    public field: string;
     public constructor(@Spec spec: ~i~this) {}
 }
+let a = new A({field: 'hello'});
+console.log(a.field); // prints: hello
 ----
 
-.Spec-style Constructor
+The code for initializing the public field of `A` is automatically generated, thanks to the `@Spec` annotation being
+given in the constructor.
+
+.@Spec Constructor
 [req,id=IDE-59,version=1]
 --
 
-1.  Annotation `@Spec` may only appear on a formal parameter of a constructor.
+1.  Annotation `@Spec` may only appear on a formal parameter of a constructor. Such a formal parameter is then called
+_@Spec parameter_ or simply _spec parameter_ and its owning constructor is referred to as a _@Spec constructor_ or
+_spec constructor_.
 2.  Only a single formal parameter of a constructor may be annotated with `@Spec`.
-3.  If a formal parameter is annotated with `@Spec`, the parameter’s type must be `pass:[~this]` or `pass:[~i~this]` (i.e. use-site structurally typed `this`).
-4.  Fields provided by the parameter, but not defined in the structural field type, are _not_ used to set fields.
-5.  Non-`public` fields explicitly added to the specparameter are copied as well.
-6.  Even if the `@Spec` annotation is used, the super constructor must be calledaccordingly.
-7.  The type of an additional member which match owned non-public fieldmust be subtype of the field’s type:
+3.  If a formal parameter is annotated with `@Spec`, the parameter’s type must be `pass:[~this]` or `pass:[~i~this]` (i.e.
+a use-site structural initializer field type of `this`, see <<structural-readWriteInit-field-typing>>).
+4.  A spec constructor will automatically initialize all its owned data fields, owned setters and data fields / setters
+from implemented interfaces that _are also part of the spec parameter's structural initializer field type_ with the data
+provided in the argument to the spec parameter.
+5.  Properties provided by an argument, but not defined in the parameter's structural initializer field type, are _not_
+used for initialization, even if a field of the same name exists in the class.
+6.  Fields explicitly added to the spec parameter, e.g. `@Spec spec: ~i~this with {name:string}`, are used for initialization
+if a non-public field of the same name exists in the class, either as an owned member of from an implemented interface.
+The type of such an additional field must be a subtype of the class's field’s type.
```

Comment:
> class's field’s type

(two different apostrophes)

Confusing. It means "the ((type of the field) of the class)"?